### ISSUE TITLE
minimatch function and generated regular expression work differently

### ISSUE THREAD

**Issue by m-shaka** [isaacs/minimatch#188]
Hi.
I found `minimatch(target, pattern)` and `minimatch.makeRe(pattern).test(target)` work differently.
I'm curious about the cause.

```typescript
import minimatch from "minimatch";

const pattern = "@src/modules/*/!(bar)/**";
const target = "@src/modules/foo/bar/a";

function main() {
  const re = minimatch.makeRe(pattern);
  if (re === false) return;
  console.log("minimatch:", minimatch(target, pattern));
  console.log("regex:", re.test(target));
}

main();
```

I'm using minimatch@5.1.2

**Comment by isaacs** [comment]
Good bug!

Fixed on 5.1.3

Note: there _are_ some cases where makeRe will not have the exact same behavior as minimatch, because the combination of negative extglobs and globstar can produce some situations that are impossible to represent in a regular expression pattern, and can only be properly tested by iterating over the path portions (which the minimatch method does).

But those should be relatively rare edge cases, and this isn't one of them, it's just a bug, where negative extglobs were anchoring to `$` as the end of the pattern, when they should be anchoring to `(?:$|\/)` (end or `/`).  That didn't cause minimatch to fail, because it iterates over path portions, so the end of the path portion _is_ the end of the string being tested, and `$` works fine.  But for a regular expression testing the entire string, this would obviously fail.

Good catch, thanks!

**Comment by m-shaka** [comment]
Thank you for your quick fix!
